app/src/main/java/unipd/se18/MignifyResolver.java

```diff
@@ -0,0 +1,111 @@
+package unipd.se18.eanresolvemodule;
+
+import android.util.Log;
+
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.JsonNode;
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * This class implements the Mignify API for EAN13 code resolve
+ * More at https://market.mashape.com/mignify/product-intelligence#
+ * @author Elia Bedin
+ */
+public class MignifyResolver implements EAN {
+
+    //The string resolved from an EAN product code
+    private String product = "";
+    //The string passed to the get method, postponed by the EAN string code
+    private static final String URL =
+            "https://mignify.p.mashape.com/gtins/v1.0/productsToGtin?gtin=";
+    //The authentication string you get from
+    //https://market.mashape.com/mignify/product-intelligence#
+    //after you subscribe to the Mignify API
+    private static final String MASHAPE_AUTH = "Nx432kOz2Wmsh83IAjUv6cUaxT7Rp1tDUbKjsnu7ueSdRFdOPK";
```

Comment:
Should this auth key be hidden?

Reply:
I did some reseach about this.
What I found is, you can't 100% hide the value of a literals string.
To the user, yes.
To people decompiling you java files, no, even if you use libraries that obscure you code.
The [way I found](https://medium.com/@anujjindal7/android-string-literals-obfuscation-in-code-6700a85d5cd1) works the best is to write this constants in a C _library_, build it with [ndk](https://developer.android.com/ndk/guides/ndk-build) and you get a .so which isn't reversible.
Since this isn't that big of a deal that at the moment the authenthication key is visible, I'll leave this like this.
But I will probably at least hide it to the user in the future.